src/providers/diagnostics.ts

```diff
@@ -17,57 +17,63 @@ export function refreshDiagnostics(
   diagnosticCollection: vsc.DiagnosticCollection,
 ): void {
   const diagnostics: vsc.Diagnostic[] = [];
+  findStaleConfigs(doc, diagnostics);
+  diagnosticCollection.set(doc.uri, diagnostics);
+}
 
+export function findStaleConfigs(
+  doc: vsc.TextDocument,
+  diagnostics?: vsc.Diagnostic[],
+): string[] {
+  const staleConfigs: string[] = [];
   for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
     const lineOfText = doc.lineAt(lineIndex);
-    const configSearch = CONFIG_NAME_WITH_QUOTES_REGEX.exec(lineOfText.text);
+    const configSearch = lineOfText.text.match(CONFIG_NAME_WITH_QUOTES_REGEX);
     if (configSearch !== null) {
       configSearch.forEach((match) => {
         const configName = match.substring(1, match.length - 1);
-        getStaleConfig(configName).forEach((config) => {
-          diagnostics.push(
+        if (getStaleConfig(configName).length > 0) {
```

Comment:
These changes are mostly just fixing an edge case where the variable name contains the actual config name as a substring (e.g. `const testGate = 'test'`)